test/helpers/powervs.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helpers
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/IBM-Cloud/power-go-client/clients/instance"
+	"github.com/IBM-Cloud/power-go-client/ibmpisession"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/pkg/cloud/services/authenticator"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/pkg/cloud/services/utils"
+)
+
+var (
+	pollingInterval                = time.Second * 30
+	powerVSInstanceDeletionTimeout = time.Minute * 10
+)
+
+const (
+	powervsInstanceDoesNotExist  = "pvm-instance does not exist"
+	powervsInstanceNotFound      = "could not be found"
+	powervsInstanceStateDeleting = "deleting"
+)
+
+// CheckPowerVSInstances verifies if the Powervs instances are deleted
+// before proceeding with the next cluster creation.
+func CheckPowerVSInstances(serviceInstanceID string) error {
+	pclient, err := getPowerVSInstanceClient(serviceInstanceID)
```

Comment:
For my better understanding, since we are trying to check all the VMs under this service instance is being deleted, So there wont be two jobs running at any time right, Because in case other will be creating and we are waiting for it be deleted?

Reply:
In the current CI flow, this is unlikely. Since we use one workspace for running the jobs at one time and there is not concurrency.

Reply:
Sure thanks for clarifying. It would be better if you can update the function comment to indicate only what this function is doing.

Reply:
Done